lib/src/components/NxTag/__tests__/NxTag.test.tsx

```diff
@@ -4,124 +4,125 @@
-    const component = mount(<><NxTag ref={ref}>Hello</NxTag></>);
-    expect(component.find('.nx-tag').getDOMNode()).toBe(ref.current);
+    const ref = React.createRef<HTMLLabelElement>(),
+        renderedEl = renderEl({ ref });
+
+    expect(ref.current).toBe(renderedEl);
   });
 
   it('correctly assigns supplied class', function() {
-    const classComponent = getShallowComponent({ className: 'foo' });
-    expect(classComponent.find('.nx-tag')).toHaveClassName('foo');
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
   it('correctly assigns supplied id', function() {
-    const idComponent = getShallowComponent({ id: 'test-id' });
-    expect(idComponent.find('.nx-tag')).toHaveProp('id', 'test-id');
+    const el = renderEl({ id: 'test-id' });
+    expect(el).toHaveAttribute('id', 'test-id');
   });
 
   it('renders the supplied text inside .nx-tag__text', function() {
-    const tagChildren = getShallowComponent({ children: 'tag text' });
-    expect(tagChildren.find('.nx-tag__text')).toHaveText('tag text');
-
-    expect(tagChildren.children().first()).toMatchSelector('label');
-    // expect(tagChildren.children().at(1)).toMatchSelector('span');
+    const el = renderEl({ children: 'tag text' });
+    expect(el).toHaveTextContent('tag text');
   });
 
   it('sets the nx-selectable-color--indigo class if no color prop is passed', function() {
-    expect(getShallowComponent().find('.nx-selectable-color--indigo')).toExist();
+    const el = renderEl();
+    expect(el).toHaveClass('nx-selectable-color--indigo');
   });
 
   it('sets the color class using the color if it is provided', function() {
-    const colorComponent = getShallowComponent({ color: 'orange' });
-    expect(colorComponent.find('.nx-selectable-color--orange')).toExist();
-  });
-
-  it('checks the children appear in the correct order, text then the icon', function() {
-    const getSelectedComponent = getShallowComponent({
-      children: 'foo bar',
-      selectedIcons: <NxFontAwesomeIcon icon={faPlusCircle} className="nx-tag__action" />
-    });
-
-    expect(getSelectedComponent.find('.nx-tag').children().first()).toMatchSelector('span');
-    expect(getSelectedComponent.find('.nx-tag').children().last()).toHaveClassName('nx-tag__action');
+    const el = renderEl({ color: 'orange' });
+    expect(el).toHaveClass('nx-selectable-color--orange');
   });
 });
 
 describe('NxSelectableTag', function() {
-  const minimumProps = { children: 'selectable tag', selected: false, onSelect: jest.fn() };
-  const getShallowComponent = enzymeUtils.getShallowComponent(NxSelectableTag, minimumProps);
+  type PropsWithRef = SelectableProps & RefAttributes<HTMLLabelElement>;
+
+  const minimalProps = { children: 'selectable tag', selected: false, onSelect: jest.fn() },
```

Comment:
Spies should only be created per-test (either in `it` or `beforeEach` blocks).  Our typical pattern for `minimalProps` is to fill in required functions with a simple noop, and then to replace it with a spy in the tests that actually need it
```suggestion
  const minimalProps = { children: 'selectable tag', selected: false, onSelect: () => {} },
```